src/Components/Artwork/Details/PrimaryLabelLine.tsx

```diff
@@ -1,27 +1,44 @@
 import { Text } from "@artsy/palette"
 import { useArtworkGridContext } from "Components/ArtworkGrid/ArtworkGridContext"
 import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { useFulfillmentOptionsForArtwork } from "Utils/Hooks/useFulfillmentOptions"
 import type { PrimaryLabelLineQuery } from "__generated__/PrimaryLabelLineQuery.graphql"
 import type { PrimaryLabelLine_artwork$data } from "__generated__/PrimaryLabelLine_artwork.graphql"
+import type { PrimaryLabelLine_me$data } from "__generated__/PrimaryLabelLine_me.graphql"
 import { type FC, useEffect } from "react"
 import { createFragmentContainer, graphql } from "react-relay"
 
 interface PrimaryLabelLineProps {
   label: string | null | undefined
   artwork?: PrimaryLabelLine_artwork$data
+  me?: PrimaryLabelLine_me$data
+}
+
+const doesShipFree = (
+  artwork: PrimaryLabelLine_artwork$data,
+  me?: PrimaryLabelLine_me$data,
+): boolean => {
+  const result = useFulfillmentOptionsForArtwork(artwork, me)
+  const { freeShippingToUserCountry, freeGlobalShipping } = result || {}
+  return !!(freeShippingToUserCountry || freeGlobalShipping)
```

Comment:
Can this logic be cleaned up a bit? there's fallbacks and boolean casts and then fallbacks on the option plucked from a fallback null object (or result)